seDataDevices.py

```diff
@@ -0,0 +1,731 @@
+            self._itemNames.append(itemName)
+
+        if offset <= (devlen - 2):
+            # Parse items which may be 2 bytes long
+            itemName = "offset{:03}_1_hex2".format(offset)
+            self[itemName] = self.hexData(data[offset: offset + 2])
+            self._itemNames.append(itemName)
+            itemName = "offset{:03}_4_int2".format(offset)
+            self[itemName] = struct.unpack("<H", data[offset: offset + 2])[0]
+            self._itemNames.append(itemName)
+
+        if offset <= (devlen - 4):
+            # Parse items which may be 4 bytes long
+            itemName = "offset{:03}_1_hex4".format(offset)
+            self[itemName] = self.hexData(data[offset: offset + 4])
+            self._itemNames.append(itemName)
+            itemName = "offset{:03}_3_float_LE".format(offset)
+            self[itemName] = struct.unpack("<f", data[offset: offset + 4])[0]
+            self._itemNames.append(itemName)
+            itemName = "offset{:03}_3_float_BE".format(offset)
+            self[itemName] = struct.unpack(">f", data[offset: offset + 4])[0]
+            self._itemNames.append(itemName)
+            itemName = "offset{:03}_4_int4".format(offset)
+            self[itemName] = struct.unpack("<L", data[offset: offset + 4])[0]
+            self._itemNames.append(itemName)
+            # Maybe it is a date?
+            try:
+                self["Date_offset{:03}".format(offset)] = self.formatDateStamp(self[itemName])
+                self["Time_offset{:03}".format(offset)] = self.formatTimeStamp(self[itemName])
+                # Other modules (eg se2graphite) become upset if Date and Time are not supplied, so take the 1st valid
+                # Date we find. Implicitly I'm guessing that offset zero will be the "real" date.
+                if "Date" not in self.keys():
+                    self["Date"] = self.formatDateStamp(self[itemName])
+                    self["Time"] = self.formatTimeStamp(self[itemName])
+            except ValueError:
+                # Apparently it is not a date!
+                self["Date_offset{:03}".format(offset)] = '{} is not a Date'.format(self[itemName])
+                self["Time_offset{:03}".format(offset)] = '{} is not a Time'.format(self[itemName])
+            self._itemNames.append("Date_offset{:03}".format(offset))
+            self._itemNames.append("Time_offset{:03}".format(offset))
+
+    @classmethod
+    def itemNames(cls):
+        # Warning.  If ParseDevice_Explorer is used on more than 1 seType data block in a single run, this will *NOT*
+        # return sensible values!
+        return cls._itemNames
+
+    @property
+    def defnLen(self):
+        return self._defnLen
+
+    @classmethod
+    def itemDefs(cls):
+        msg = ["\n\n{} / {} parses data blocks with any unrecognised seType.\n{}".format(cls.__name__, cls._devType, "="*80)]
+        msg.append("Each pair/quadruple of bytes is parsed multiple times, using a number of different fields types.")
```

Comment:
nitty comment: in general it's cleaner and marginally more efficient to use either a multiline string or single strings on multiple lines.  https://waymoot.org/home/python_string/

```
x = """ \
lots
of
text """

y = "lots\n" \
"of\n" \
"text"
```